sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/options/ShareFileRenameOptions.java

```diff
@@ -170,4 +174,46 @@ public ShareFileRenameOptions setMetadata(Map<String, String> metadata) {
         this.metadata = metadata;
         return this;
     }
+
+    /**
+     * Gets the {@link ShareFileHttpHeaders}. Currently, only content type is respected. Others are ignored.
+     *
+     * @return The {@link ShareFileHttpHeaders}.
+     */
+    public ShareFileHttpHeaders getHeaders() {
+        return this.headers;
+    }
+
+    /**
+     * Sets the {@link ShareFileHttpHeaders}.  Currently, only content type is respected. This method will throw if
+     * others are set.
+     *
+     * @param headers {@link ShareFileHttpHeaders}
+     * @return The updated options.
+     * @throws IllegalArgumentException If headers besides content type are set, this method will throw.
+     */
+    public ShareFileRenameOptions setHeaders(ShareFileHttpHeaders headers) {
+        if (headers.getCacheControl() != null) {
+            throw LOGGER.logExceptionAsError(
+                new IllegalArgumentException("cache control is not supported on this api"));
+        }
+        if (headers.getContentDisposition() != null) {
+            throw LOGGER.logExceptionAsError(
+                new IllegalArgumentException("content disposition is not supported on this api"));
+        }
+        if (headers.getContentEncoding() != null) {
+            throw LOGGER.logExceptionAsError(
+                new IllegalArgumentException("content encoding is not supported on this api"));
+        }
+        if (headers.getContentLanguage() != null) {
+            throw LOGGER.logExceptionAsError(
+                new IllegalArgumentException("content language is not supported on this api"));
+        }
+        if (headers.getContentMd5() != null) {
+            throw LOGGER.logExceptionAsError(
+                new IllegalArgumentException("content md5 is not supported on this api"));
+        }
+        this.headers = headers;
+        return this;
```

Comment:
let's merge it. but we should discuss this issue during api review.

Reply:
I vote we make ContentType a property on ShareFileRenameOptions